contrib_bots/lib/sched.py

```diff
@@ -0,0 +1,211 @@
+# pip install schedule is necessary
+import schedule
+import time
+import json
+import threading
+import os
+import uuid
+from random import randint
+
+config = os.path.expanduser('~/.sched')
+with open(config) as data_file:
+    wakeup = json.load(data_file)[0]
+
+cache = os.path.expanduser('~/.cache/sched')
+# Verify is hour format is valid
+def is_time_format(format):
+    try:
+        time.strptime(format, '%H:%M')
+        return True
+    except ValueError:
+        return False
+
+# For every task we want to schedule we define a job that have a random unique id  to cancel the task.
+
+def job(randomint, message, client):
+    original_content = message['content']
+    reduced_message_list = original_content.split()[3:]
+    reduced_message_string = " ".join(reduced_message_list)
+    if reduced_message_string:
+        if reduced_message_string.split()[0] == "at":
+            reduced_message_string = reduced_message_string.split()[2:]
+            reduced_message_string = " ".join(reduced_message_string)
+
+    events = dict(
+        type='stream',
+        to='alert',
+        subject="All alerts",
+        content=reduced_message_string+" "+randomint,
+        )
+    client.send_message(events)
+
+def start_up(client):
+    events = dict(
+        type='stream',
+        to='alert',
+        subject="All alerts",
+        content="@recover 0",
+        )
+    client.send_message(events)
+    return schedule.CancelJob
+
+class SchedHandler(object):
+    '''
+    This bot allows you to schedule daily alerts. It
+    looks for messages starting with '@every' or '@cancel' or '@recover'.
+    In this example, we write alerts at certain time intervals
+    in a new stream called "alert,".
+    '''
+    # Now you can add a message when your bot starts
+    def start_up(self, client, state_handler):
+        time.sleep(wakeup["timeout_in_seconds"]+7)
+        schedule.every(1).seconds.do(start_up, client)
+
+    def usage(self):
+        def run_threaded(job_func):
+            job_thread = threading.Thread(target=job_func)
+            job_thread.start()
+
+        def event_handler():
+            time.sleep(wakeup["timeout_in_seconds"]+4)
+            schedule.run_pending()
+        schedule.every(wakeup["timeout_in_seconds"]).seconds.do(run_threaded, event_handler)
+        time.sleep(wakeup["timeout_in_seconds"]+1)
+        schedule.run_pending()
```

Comment:
From the `schedule` example:
```Python
import schedule
import time

def job():
    print("I'm working...")

schedule.every(10).minutes.do(job)
schedule.every().hour.do(job)
schedule.every().day.at("10:30").do(job)
schedule.every().monday.do(job)
schedule.every().wednesday.at("13:15").do(job)

while True:
    schedule.run_pending()
    time.sleep(1)
```

You need to run
```
while True:
    schedule.run_pending()
    time.sleep(1)
```
in a separate thread. Initialize that thread in `__init__`.